common/achievement/get.ts

```diff
@@ -156,6 +162,9 @@ const assembleAchievementData = async (userAchievements: achievements_with_templ
         where: { group: userAchievements[currentAchievementIndex].template.group, isActive: true },
         orderBy: { groupOrder: 'asc' },
     });
+    const derivedTemplates = deriveAchievementTemplates(userAchievements[currentAchievementIndex].template.group);
+    achievementTemplates.push(...derivedTemplates);
+    achievementTemplates.sort((left, right) => left.groupOrder - right.groupOrder);
```

Comment:
this looks strange - shouldn't it sort by group AND groupOrder? Or why by groupOrder?

Reply:
Because in this case all achievements in the list are of the same group